lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx

```diff
@@ -255,6 +256,98 @@ Interval LinearModelAnalysis::getCoefficientsConfidenceInterval(const Scalar lev
   return bounds;
 }
 
+/* Check that the sample size is large enough to compute asymptotic distributions */
+void LinearModelAnalysis::checkSampleSize() const
+{
+  const UnsignedInteger size = linearModelResult_.getInputSample().getSize();
+  const UnsignedInteger minSize = ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-MinimumSampleSizeForAsymptoticDistributions");
+  if (!(size >= minSize))
+    throw InvalidArgumentException(HERE) << "The sample size is too small (must be at least " << minSize << ")";
+}
+
+SymmetricMatrix LinearModelAnalysis::getGramInverse()
+{
+  if (!gramInverseComputed_)
+  {
+    gramInverse_ = linearModelResult_.getDesign().computeGram().inverse();
+    gramInverseComputed_ = true;
```

Comment:
we should avoid such cache pattern which does not play well in a multithread context, at least in new code